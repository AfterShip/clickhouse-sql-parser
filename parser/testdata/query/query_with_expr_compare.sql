SELECT date, path, splitByChar('/', path)[2] AS path_b
FROM(
    SELECT 'pathA/pathB/pathC' AS path, '2024-09-10' AS date
    )
WHERE toDate(date) BETWEEN '2024-09-01' AND '2024-09-30'
  AND splitByChar('/', path)[1] = 'pathA'