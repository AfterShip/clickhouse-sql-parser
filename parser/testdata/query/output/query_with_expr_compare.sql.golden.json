[
  {
    "SelectPos": 0,
    "StatementEnd": 225,
    "With": null,
    "Top": null,
    "HasDistinct": false,
    "SelectItems": [
      {
        "Expr": {
          "Name": "date",
          "QuoteType": 1,
          "NamePos": 7,
          "NameEnd": 11
        },
        "Modifiers": [],
        "Alias": null
      },
      {
        "Expr": {
          "Name": "path",
          "QuoteType": 1,
          "NamePos": 13,
          "NameEnd": 17
        },
        "Modifiers": [],
        "Alias": null
      },
      {
        "Expr": {
          "Object": {
            "Name": {
              "Name": "splitByChar",
              "QuoteType": 1,
              "NamePos": 19,
              "NameEnd": 30
            },
            "Params": {
              "LeftParenPos": 30,
              "RightParenPos": 40,
              "Items": {
                "ListPos": 32,
                "ListEnd": 40,
                "HasDistinct": false,
                "Items": [
                  {
                    "Expr": {
                      "LiteralPos": 32,
                      "LiteralEnd": 33,
                      "Literal": "/"
                    },
                    "Alias": null
                  },
                  {
                    "Expr": {
                      "Name": "path",
                      "QuoteType": 1,
                      "NamePos": 36,
                      "NameEnd": 40
                    },
                    "Alias": null
                  }
                ]
              },
              "ColumnArgList": null
            }
          },
          "Params": {
            "LeftBracketPos": 41,
            "RightBracketPos": 43,
            "Items": {
              "ListPos": 42,
              "ListEnd": 43,
              "HasDistinct": false,
              "Items": [
                {
                  "Expr": {
                    "NumPos": 42,
                    "NumEnd": 43,
                    "Literal": "2",
                    "Base": 10
                  },
                  "Alias": null
                }
              ]
            }
          }
        },
        "Modifiers": [],
        "Alias": {
          "Name": "path_b",
          "QuoteType": 1,
          "NamePos": 48,
          "NameEnd": 54
        }
      }
    ],
    "From": {
      "FromPos": 55,
      "Expr": {
        "Table": {
          "TablePos": 59,
          "TableEnd": 121,
          "Alias": null,
          "Expr": {
            "HasParen": true,
            "Select": {
              "SelectPos": 65,
              "StatementEnd": 121,
              "With": null,
              "Top": null,
              "HasDistinct": false,
              "SelectItems": [
                {
                  "Expr": {
                    "LiteralPos": 73,
                    "LiteralEnd": 90,
                    "Literal": "pathA/pathB/pathC"
                  },
                  "Modifiers": [],
                  "Alias": {
                    "Name": "path",
                    "QuoteType": 1,
                    "NamePos": 95,
                    "NameEnd": 99
                  }
                },
                {
                  "Expr": {
                    "LiteralPos": 102,
                    "LiteralEnd": 112,
                    "Literal": "2024-09-10"
                  },
                  "Modifiers": [],
                  "Alias": {
                    "Name": "date",
                    "QuoteType": 1,
                    "NamePos": 117,
                    "NameEnd": 121
                  }
                }
              ],
              "From": null,
              "ArrayJoin": null,
              "Window": null,
              "Prewhere": null,
              "Where": null,
              "GroupBy": null,
              "WithTotal": false,
              "Having": null,
              "OrderBy": null,
              "LimitBy": null,
              "Limit": null,
              "Settings": null,
              "Format": null,
              "UnionAll": null,
              "UnionDistinct": null,
              "Except": null
            }
          },
          "HasFinal": false
        },
        "StatementEnd": 121,
        "SampleRatio": null,
        "HasFinal": false
      }
    },
    "ArrayJoin": null,
    "Window": null,
    "Prewhere": null,
    "Where": {
      "WherePos": 128,
      "Expr": {
        "LeftExpr": {
          "Expr": {
            "Name": {
              "Name": "toDate",
              "QuoteType": 1,
              "NamePos": 134,
              "NameEnd": 140
            },
            "Params": {
              "LeftParenPos": 140,
              "RightParenPos": 145,
              "Items": {
                "ListPos": 141,
                "ListEnd": 145,
                "HasDistinct": false,
                "Items": [
                  {
                    "Expr": {
                      "Name": "date",
                      "QuoteType": 1,
                      "NamePos": 141,
                      "NameEnd": 145
                    },
                    "Alias": null
                  }
                ]
              },
              "ColumnArgList": null
            }
          },
          "Between": {
            "LiteralPos": 156,
            "LiteralEnd": 166,
            "Literal": "2024-09-01"
          },
          "AndPos": 168,
          "And": {
            "LiteralPos": 173,
            "LiteralEnd": 183,
            "Literal": "2024-09-30"
          }
        },
        "Operation": "AND",
        "RightExpr": {
          "LeftExpr": {
            "Object": {
              "Name": {
                "Name": "splitByChar",
                "QuoteType": 1,
                "NamePos": 191,
                "NameEnd": 202
              },
              "Params": {
                "LeftParenPos": 202,
                "RightParenPos": 212,
                "Items": {
                  "ListPos": 204,
                  "ListEnd": 212,
                  "HasDistinct": false,
                  "Items": [
                    {
                      "Expr": {
                        "LiteralPos": 204,
                        "LiteralEnd": 205,
                        "Literal": "/"
                      },
                      "Alias": null
                    },
                    {
                      "Expr": {
                        "Name": "path",
                        "QuoteType": 1,
                        "NamePos": 208,
                        "NameEnd": 212
                      },
                      "Alias": null
                    }
                  ]
                },
                "ColumnArgList": null
              }
            },
            "Params": {
              "LeftBracketPos": 213,
              "RightBracketPos": 215,
              "Items": {
                "ListPos": 214,
                "ListEnd": 215,
                "HasDistinct": false,
                "Items": [
                  {
                    "Expr": {
                      "NumPos": 214,
                      "NumEnd": 215,
                      "Literal": "1",
                      "Base": 10
                    },
                    "Alias": null
                  }
                ]
              }
            }
          },
          "Operation": "=",
          "RightExpr": {
            "LiteralPos": 220,
            "LiteralEnd": 225,
            "Literal": "pathA"
          },
          "HasGlobal": false,
          "HasNot": false
        },
        "HasGlobal": false,
        "HasNot": false
      }
    },
    "GroupBy": null,
    "WithTotal": false,
    "Having": null,
    "OrderBy": null,
    "LimitBy": null,
    "Limit": null,
    "Settings": null,
    "Format": null,
    "UnionAll": null,
    "UnionDistinct": null,
    "Except": null
  }
]